docs/sdkx-cocos2dx/users.mdx

```diff
@@ -0,0 +1,294 @@
+
+#### How to use the failure delegates
+
+If identity verification fails, the SDK will invoke the `authenticationFailed` delegate to notify the application of the failure:
+
+<table>
+  <tr>
+    <th>Authentication failure reason</th>
+    <th>When is it called</th>
+    <th>When to use</th>
+  </tr>
+  <tr>
+    <td>
+      HSAuthenticationFailureReason
+      <br />
+      .HSAuthTokenNotProvided
+    </td>
+    <td>When no 'user authentication token' is provided</td>
+    <td>
+      If you do not plan on sending a 'user authentication token' when
+      implementing SDK, but plan on implementing it in the future, you can use
+      this failure delegate to show your own alerts to the user, such as a
+      prompt to update the app. You may want to use this if you are using the
+      login API without the 'user authentication token', since these users will
+      be considered unverified once 'Identity Verification' is enabled on the
+      Dashboard. Using this delegate is completely optional - Helpshift will
+      prevent unverified users from being able to file Issues, as mentioned
+      previously.
+    </td>
+  </tr>
+  <tr>
+    <td>
+      HSAuthenticationFailureReason
+      <br />
+      .HSInvalidAuthToken
+    </td>
+    <td>When the 'user authentication token' is invalid</td>
+    <td>
+      If the HMAC Digest being provided via LoginAPI is invalid, then Helpshift
+      will prevent the users from filing Issues. The 'user authentication token'
+      can be invalid if:
+      <ul>
+        <li>
+          You made a programming error. Check how to generate HMAC Digest{" "}
+          <a href="https://support.helpshift.com/kb/article/how-do-i-configure-the-endpoint-and-my-app-web-chat-widget-for-user-identity-verification/">
+            here
+          </a>
+          .
+        </li>
+        <li>
+          You regenerated the secret key on the Dashboard, but didn't update the
+          'user authentication token'.
+        </li>
+        <li>A 3rd party is trying to make requests.</li>
+      </ul>
+      When the 'user authentication token' is invalid, end-users will not able
+      to file Issues and an error is shown to them, as mentioned previously. You
+      can use this delegate if you want to show your own alerts or re-fetch the
+      correct auth token from your server.
+    </td>
+  </tr>
+</table>
+
+The authentication failure reason is wrapped in an enum `HSAuthenticationFailureReason`.
+
+Example:
+
+```cpp
+
+void handleHelpshiftEvent(const char *eventName, cocos2d::ValueMap &eventData) {
+    // handle Helpshift events
+    ...
+}
+void handleAuthFailureEvent(HSAuthenticationFailureReason reason) {
+    if (reason == HSAuthenticationFailureReason::HSAuthTokenNotProvided) {
+        ...
+    } else if (reason == HSAuthenticationFailureReason::HSInvalidAuthToken) {
+        ...
+    }
+}
+HelpshiftCocos2dx::setHelpshiftEventsListener(handleHelpshiftEvent,
```

Comment:
i think this line is missing in all other places where we have used handleHelpshiftEvent or handleAuthFailureEvent.
Please check if we should be using this line at all other places for clarity in code samples